src/NHSD.BuyingCatalogue.Solutions.Persistence/Repositories/PriceRepository.cs

```diff
@@ -31,7 +31,7 @@ public sealed class PriceRepository : IPriceRepository
                 INNER JOIN dbo.PricingUnit AS pu ON pu.PricingUnitId = cp.PricingUnitId
                 INNER JOIN dbo.CatalogueItem AS ci ON ci.CatalogueItemId = cp.CatalogueItemId
                 LEFT OUTER JOIN dbo.CataloguePriceTier AS cptr ON cptr.CataloguePriceId = cp.CataloguePriceId
-        WHERE   cp.CatalogueItemId = @solutionId;";
+        WHERE   cp.CatalogueItemId = @catalogueItemId;";
```

Comment:
Have a look at the `SupplierRepository` class, @bjss-chris has used a nice convention to basically ignore a parameter passed into an SQL query.

e.g.
`AND c.PublishedStatusId = ISNULL(NULLIF(@statusId, ''), c.PublishedStatusId)`

This could be used to retrieve all prices if no catalogueItemId is supplied to the List prices query. Something like the following:

`WHERE cp.CatalogueItemId  = ISNULL(NULLIF(@catalogueItemId, ''), c.CatalogueItemId)`